content/toolsAndResources/nge.md

```diff
@@ -0,0 +1,126 @@
+
+The tool is available as a stand alone experience by visiting [Node Geometry Editor](https://nge.babylonjs.com) and can also be opened from the inspector when selecting any Node Geometry in scene.
+
+![Node Geometry Editor Launch Button](/img/tools/nge/ngeButton.jpg)
+
+## Interface
+
+The interface for the editor is broken into 4 main sections as can be seen below. The node list is the entire list of available nodes that can be dragged onto the graph. At the top of the list is a search box to help narrow the list. The graph is the workspace for arranging the nodes to create the geometry flow. There is a panel for parameters and controls on the right side of the window which displays any parameters available to the currently selected node. If nothing on the graph is selected, this panel shows controls for the tool itself to load, save, or change default settings. In the bottom right is the preview window which displays the current Node Geometry output. Lastly, at the bottom of the window is the console log panel which gives information the build status of the current graph. 
+
+![Babylon.js Node Geometry Editor](/img/tools/nge/ngeWindow.jpg)
+
+The image above shows three types of nodes available in the graph, an input node, a source node, and an output node. At minimum all graphs need exactly one output node and at least one source node. Note that the inputs and outputs on the nodes share colors and icons that give hints on what can be connected. Since the `Geometry Output` node has only a geometry input, there needs to be at least one node supplying that that input.
+
+Input is just one type of several other node types such as contextual, logical, math, and more. Connecting these different node types to the source nodes can produce an almost limitless variety of meshes. 
+
+## Adding Nodes
+
+Adding nodes to the graph is one of the most common actions taken while working with the editor. There are two main methods for adding node to the graph. The first method is clicking and dragging a node from the node list onto the graph and releasing the mouse button. This will drop the selected node at the mouse position. The node list can be filtered by entering a keyword in the search bar at the top of the list to help find a node faster.
+
+The second method is to press the spacebar for a shortcut to open a search window with a the full node list at the mouse position on the graph. Once the window is open, we can simply start typing to enter a keyword in the search bar. Once the list is filtered, use either the mouse to click the desired node or use the arrow keys to highlight the desired node and press enter. This will drop the node at the mouse position when the shortcut was invoked. 
+
+![Press spacebar for shortcut to search bar to filter node list which appears at the mouse position](/img/tools/nge/nodeAddShortcut.jpg)

```

Comment:
Do we want to mention that if a link is selected the new node will be added in that link?

Reply:
We certainly can add that context. I will update.

Reply:
In testing for this update, if a wire is selected and then the shortcut is invoked, but the user selects an incompatible node - say an AND node when a Vector3 wire is selected - the new node will not be connected, but the original connection will also be severed. It seems like we should update the behavior to only alter the original connection if the new node is compatible with the data type represented by the wire.

Reply:
in latest commit to PR